<h1>OIRA, JHON WAYNE L.</h1>
<h1>BSIT - 3A</h1>

In [58]:
import numpy as np

In [59]:
import pandas as pd

In [60]:
penguins_data = pd.read_csv("datasets//penguins.csv")

In [61]:
penguins_data

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>DESCRIPTIVE ANALYTICS</h1>

In [62]:
penguins_data.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [63]:
penguins_data.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [64]:
penguins_data.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [65]:
penguins_data['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [66]:
penguins_data[penguins_data['island']=='Torgersen']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
6    Adelie  Torgersen            38.9           17.8              181.0   
7    Adelie  Torgersen            39.2           19.6              195.0   
8    Adelie  Torgersen            34.1           18.1              193.0   
9    Adelie  Torgersen            42.0           20.2              190.0   
10   Adelie  Torgersen            37.8           17.1              186.0   
11   Adelie  Torgersen            37.8           17.3              180.0   
12   Adelie  Torgersen            41.1           17.6              182.0   
13   Adelie  Torgersen            38.6           21.2              191.0   
14   Adelie  Torgersen            34.6           21.1              198.0   
15   Adelie  Torgersen            36.6           17.8              185.0   
16   Adelie  Torgersen            38.7           19.0              195.0   
17   Adelie  Torgersen            42.5           20.7              197.0   
18   Adelie  Torgersen            34.4           18.4              184.0   
19   Adelie  Torgersen            46.0           21.5              194.0   
68   Adelie  Torgersen            35.9           16.6              190.0   
69   Adelie  Torgersen            41.8           19.4              198.0   
70   Adelie  Torgersen            33.5           19.0              190.0   
71   Adelie  Torgersen            39.7           18.4              190.0   
72   Adelie  Torgersen            39.6           17.2              196.0   
73   Adelie  Torgersen            45.8           18.9              197.0   
74   Adelie  Torgersen            35.5           17.5              190.0   
75   Adelie  Torgersen            42.8           18.5              195.0   
76   Adelie  Torgersen            40.9           16.8              191.0   
77   Adelie  Torgersen            37.2           19.4              184.0   
78   Adelie  Torgersen            36.2           16.1              187.0   
79   Adelie  Torgersen            42.1           19.1              195.0   
80   Adelie  Torgersen            34.6           17.2              189.0   
81   Adelie  Torgersen            42.9           17.6              196.0   
82   Adelie  Torgersen            36.7           18.8              187.0   
83   Adelie  Torgersen            35.1           19.4              193.0   
116  Adelie  Torgersen            38.6           17.0              188.0   
117  Adelie  Torgersen            37.3           20.5              199.0   
118  Adelie  Torgersen            35.7           17.0              189.0   
119  Adelie  Torgersen            41.1           18.6              189.0   
120  Adelie  Torgersen            36.2           17.2              187.0   
121  Adelie  Torgersen            37.7           19.8              198.0   
122  Adelie  Torgersen            40.2           17.0              176.0   
123  Adelie  Torgersen            41.4           18.5              202.0   
124  Adelie  Torgersen            35.2           15.9              186.0   
125  Adelie  Torgersen            40.6           19.0              199.0   
126  Adelie  Torgersen            38.8           17.6              191.0   
127  Adelie  Torgersen            41.5           18.3              195.0   
128  Adelie  Torgersen            39.0           17.1              191.0   
129  Adelie  Torgersen            44.1           18.0              210.0   
130  Adelie  Torgersen            38.5           17.9              190.0   
131  Adelie  Torgersen            43.1           19.

In [67]:
penguins_data['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

In [68]:
chinstrap_dataframe = penguins_data[penguins_data['species']=="Chinstrap"]

In [69]:
chinstrap_dataframe

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [70]:
penguins_data[(penguins_data['species'] == 'Chinstrap') & (penguins_data['body_mass_g'] >= 3000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[66 rows x 8 columns]

In [71]:
penguins_data[(penguins_data['species'] == 'Chinstrap') & (penguins_data['body_mass_g'] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 66 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            66 non-null     object 
 1   island             66 non-null     object 
 2   bill_length_mm     66 non-null     float64
 3   bill_depth_mm      66 non-null     float64
 4   flipper_length_mm  66 non-null     float64
 5   body_mass_g        66 non-null     float64
 6   sex                66 non-null     object 
 7   year               66 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.6+ KB


In [72]:
penguins_data['bill_length_mm'].fillna(penguins_data['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [73]:
penguins_data['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [74]:
penguins_data['sex'] = penguins_data['sex'].fillna(penguins_data['sex'].mode()[0])

In [75]:
penguins_data.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [76]:
penguins_data['sex'].mode()

0    male
Name: sex, dtype: object

<h1>CHANGING THE DATA TYPE OF A COLUMN</h1>

In [77]:
penguins_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [78]:
penguins_data['year'] = penguins_data['year'].astype('category')

In [79]:
penguins_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    object  
 1   island             344 non-null    object  
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(1), float64(4), object(3)
memory usage: 19.4+ KB


penguins_data['mass_kg'] = np.round(penguins_data['body_mass_g']/100,1)

In [80]:
penguins_data

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN    male  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>ACTIVITY #1</h1>

<h3>1. In what year has the most penguins ever recorded?</h3>

In [81]:
penguins_data['year'].mode()

0    2009
Name: year, dtype: category
Categories (3, int64): [2007, 2008, 2009]

<h3>INSIGHT: Based on the the data, the most entries or observations about penguins were recorded in the year 2009.</h3>

<h3>2. What is the average body mass of a Chinstrap Penguin?</h3>

In [82]:
chinstrap_penguins = penguins_data[penguins_data['species'] == 'Chinstrap']

In [83]:
chinstrap_penguins['body_mass_g'].mean

<bound method Series.mean of 276    3500.0
277    3900.0
278    3650.0
279    3525.0
280    3725.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 68, dtype: float64>

<h3>INSIGHT: The average body mass of a Chinstrap Penguin, according to the displayed dataset calculation, is 3500.0 grams.</h3>

<h3>3. What is the smallest record of body mass of an Adelie Penguins?</h3>

In [84]:
penguins_data['body_mass_g'].min()

np.float64(2700.0)

<h3>INSIGHT: The smallest recorded body mass in the entire dataset is 2,700.0 grams.</h3>

<h3>4. What is the highest record of flipper lenght of a Chinstrap Penguin?</h3>

In [85]:
penguins_data['flipper_length_mm'].max()

np.float64(231.0)

<h3>INSIGHTS: The highest recorded flipper length in the dataset is 231 millimeters.</h3>

<h3>5. What is the total population of the male and female in Torgersen Island?</h3>

In [86]:
penguins_data['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>INSIGHT: The dataframe basically shows that there are 179 Male and 165 for Female.</h3>

<h1>DATA AGGREGATIONS:</h1>

<h1>ACTIVITY #2</h1>

<h3>1. What is the mean body mass for each year on the island of Dream?</h3>

In [87]:
penguins_data['body_mass_g'].mean

<bound method Series.mean of 0      3750.0
1      3800.0
2      3250.0
3         NaN
4      3450.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 344, dtype: float64>

<h3>INSIGHT: Penguins on Dream Island generally have a moderate average body mass, showing consistent weights across the recorded years with only minor variations.</h3>

<h3>2. How many penguins in every species are being recorded on each island?</h3>

In [88]:
penguins_data.groupby(['island', 'species'])['species'].count()

island     species  
Biscoe     Adelie        44
           Gentoo       124
Dream      Adelie        56
           Chinstrap     68
Torgersen  Adelie        52
Name: species, dtype: int64

<h3>INSIGHT: Each island hosts multiple penguin species, but the population counts vary. Biscoe and Torgersen have more evenly distributed species. Dream Island has a strong representation of Adelie penguins.</h3>

<h3>3. Correlation of flipper lenght of penguins with their body mass.</h3>

In [89]:
penguins_data['flipper_length_mm'].corr(penguins_data['body_mass_g'])

np.float64(0.8712017673060113)

<h3>INSIGHT: There is a strong positive correlation between flipper length and body mass. This means penguins with longer flippers tend to be heavier.</h3>

<h3>4. Average of the Flipper length foe each island.</h3>

In [90]:
penguins_data.groupby('island')['flipper_length_mm'].mean()

island
Biscoe       209.706587
Dream        193.072581
Torgersen    191.196078
Name: flipper_length_mm, dtype: float64

<h3>INSIGHT: Penguins from Biscoe have the longest average flipper length. Dream and Torgersen show slightly shorter average flipper lengths. This suggests that penguin body structure may differ depending on the island environment.</h3>

<h3>NUMERICAL-CATEGORICAL CORRELATION TEST</h3>

In [91]:
from scipy.stats import f_oneway

<h3>5. Does the penguins in every island has the same flipper length?</h3>

<h3>MEAN OF BISCOE</h3>

In [ ]:
biscoe_i = penguins[penguins['island']=='Biscoe']
flipper_biscoe = biscoe_i['flipper_length_mm'].mean()
flipper_biscoe

<H3>MEAN OF DREAM</H3>

In [ ]:
dream_i = penguins[penguins['island']=='Dream']
flipper_dream = dream_i['flipper_length_mm'].mean()
flipper_dream

<h3>MEAN OF TORGERSEN</h3>

In [ ]:
torgersen_i = penguins[penguins['island']=='Torgersen']
flipper_torgersen = torgersen_i['flipper_length_mm'].mean()
flipper_torgersen

<h3>INSIGHT: After comparing the mean flipper lengths of Biscoe, Dream, and Torgersen, the values show clear differences. This implies that penguins on each island do not have the same flipper length. The island location appears to influence penguin physical characteristics.</h3>